documentation/docs/04-advanced-routing.md

```diff
@@ -47,7 +51,25 @@ src/routes/
 └ +error.svelte
 ```
 
-> `src/routes/a/[...rest]/z/+page.svelte` will match `/a/z` (i.e. there's no parameter at all) as well as `/a/b/z` and `/a/b/c/z` and so on. Make sure you check that the value of the rest parameter is valid, for example using a [matcher](#matching).
+```js
+/// file: src/routes/marx-brothers/[...path]/+page.js
+import { error } from '@sveltejs/kit';
+
+/** @type {import('./$types').PageLoad} */
+export function load(event) {
+	throw error(404, 'Not Found');
+}
+```
+
+Alternatively, you could also `throw` a `redirect` to get back to an existing page.
+
+If you don't want to invoke `+error.svelte` in this case, you could instead create a `+page.svelte` file with your custom 404 content:
+
+```html
+/// file: src/routes/marx-brothers/[...path]/+page.svelte
+<h1>Not Found</h1>
+<p>This Marx brother never lived</p>
+```
```

Comment:
Redirecting in this situation is a bit of a bad practice, and it's very bad for SEO to return a 200 for a not found resource, so I don't think we should promote these patterns in the docs

```suggestion
```

Reply:
That makes sense

Reply:
On second thought - why is `redirect` bad practise? It's the only way to have it working for both client and server (doing it in `handle` only works for the server, SPAs are left out), and in case of a first page hit the browser will just get a redirect response just like it would from the `handle` hook.

Reply:
It's just semantically incorrect. HTTP status codes have very specific meanings, and none of the [redirection codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#redirection_messages) (300, 301, 302, 303, 304, 307, 308) are appropriate when a resource can't be found

Reply:
Internal SPAs don't care about semantics, but I agree that this is a good enough argument to not bless it in the docs - people will likely try out and see that works, too.